### PR TITLE
Build wheels on the 1st and 15th of every month

### PR DESCRIPTION
**Pull request by Saransh-cpp**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: .github/workflows/publish_pypi.yml [hunk 1]
```diff
@@ -3,6 +3,9 @@ name: Build and publish package to PyPI
 on:
```
Comment (Saransh-cpp) [inline .github/workflows/publish_pypi.yml:3]: Should we add a badge for this in the README?

Comment (agriyakhetarpal) [reply .github/workflows/publish_pypi.yml:3]: Should be a good idea. Alternatively or in addition we can use one of those actions to open an issue if one of the jobs fails

Comment (Saransh-cpp) [reply .github/workflows/publish_pypi.yml:3]: Works now - https://github.com/Saransh-cpp/PyBaMM/issues/13!

**Review by agriyakhetarpal — APPROVED** [review]
Thanks, @Saransh-cpp!

**Review by brosaplanella — APPROVED** [review]
Looks great, thanks!

**Pull request merged**